Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -2092,6 +2094,42 @@ - (void)didTapContactShareViewItem:(ConversationViewItem *)conversationItem
     [self.navigationController pushViewController:view animated:YES];
 }
 
+- (void)sendMessageToContactShare:(ContactShareViewModel *)contactShare
+{
+    OWSAssertIsOnMainThread();
+    OWSAssert(contactShare);
+
+    self.contactShareViewHelper = [[ContactShareViewHelper alloc] initWithContactShare:contactShare
+                                                                       contactsManager:self.contactsManager
+                                                                    fromViewController:self
+                                                                              delegate:self];
+    [self.contactShareViewHelper sendMessageToContact];
```

Comment:
It's unfortunate but we need to retain a strong reference to `ContactShareViewHelper` while we're using it.

Reply:
It's a tough choice, because we either need "the last helper" (as we have it now) or we'll need the helper to know "the last contact share".  I prefer the former since we're using delegate pattern with system UI.  If we start a new action, we create a new helper which ensures that the old helper will be deallocated.  If we used a single mutable helper, we could conceivably receive a delegate invocation related to a different contact share.

Reply:
I was thinking like this: https://github.com/signalapp/Signal-iOS/pull/3398